nbconvert/preprocessors/execute.py

```diff
@@ -56,11 +63,77 @@ def preprocess(self, nb, resources=None, km=None):
         resources : dictionary
             Additional resources used in the conversion process.
         """
-        # Copied from NotebookClient init :/
-        self.nb = nb
-        self.km = km
-        if resources:
-            self.resources = resources
-        self.reset_execution_trackers()
+        # This may look wrong, but preprocess acts like an init on execution state and
```

Comment:
Let's move all of this into docstring with an important directive so that this populates through to the docs.